mantidproject/mantid
# Replace boost end_with and starts_with by C++20 std::string built in functions

### Description of work
Since our move to C++20, the string functions `starts_with` and `ends_with` are now available as part of `std::string`. Here we replace all instances of the boost equivalents. Note that the case insensitive versions (`istarts_with` and `iends_with`) have not been replaced because there isn't a suitable STL alternative.

#### Summary of work
Things that had to be done as part of this work:
- replace the boost `ends_with` and `starts_with` with the STL versions
- remove the headers from places where they're not needed
- add headers to places where they are needed
- fix cppcheck defects where the lines numbers have changed w.r.t. the suppressions file

*There is no associated issue.*

### To test:
Check that the changes are all sensible. Check the codebase for any missed instances of `boost::algorithm::starts_with` or `boost::algorithm::ends_with`.


*This does not require release notes* because **it's code maintenance.**



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
